server/services/waypoint.js

```diff
@@ -358,7 +364,17 @@ module.exports = class WaypointService {
     }
 
     performWaypointActionsCollects(game, waypoints) {
-        this._performFilteredWaypointActions(game, waypoints, ['collectAll', 'collect', 'collectAllBut']);
+        this._performFilteredWaypointActions(game, waypoints, ['collectAll', 'collect', 'collectAllBut', 'collectPercentage']);
+    }
+
+    _performWaypointActionDropPercentage(carrier, star, waypoint) {
+        const toDrop = Math.floor(carrier.ships * (waypoint.actionShips * 0.01))
+
+        if (toDrop > 1 && carrier.ships - toDrop > 0) {
```

Comment:
Just to confirm, this will ensure that the carrier will always have at least 1 ship?

Reply:
Yes, I've changed it now to use >= though as that makes the intention more clear.